geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -1118,6 +1115,18 @@ public void distribute(EnumListenerEvent operation, EntryEventImpl event,
     }
   }
 
+  private void recordDroppedEvent(EntryEventImpl event) {
+    final boolean isDebugEnabled = logger.isDebugEnabled();
```

Comment:
This line could be moved to immediately before the `if (isDebugEnabled)` check so that it's only executed if necessary rather than unnecessarily in the case that `this.eventProcessor != null` evaluates to true.